### PR TITLE
chore: add github workflow that runs go vet

### PR DESCRIPTION
**Pull request by efekarakus**
- Run [go vet](https://golang.org/cmd/vet/) on all of our PRs.
- Fix two issues that go vet found.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by efekarakus** [comment]
In the long run it'll be cool to use this github action: https://github.com/golangci/golangci-lint-action (https://golangci-lint.run/usage/install/) but there are too many issues that needs to be fixed :) so started with this to lay the groundwork.

File: internal/pkg/cli/task_run.go [hunk 1]
```diff
@@ -311,7 +311,7 @@ func (o *runTaskOpts) validateFlagsWithSubnets() error {
 	}
 
 	if o.useDefaultSubnets {
-		fmt.Errorf("cannot specify both `--subnets` and `--default`")
+		return fmt.Errorf("cannot specify both `--subnets` and `--default`")
```
Comment (iamhopaul123) [inline internal/pkg/cli/task_run.go:314]: wut...

File: internal/pkg/manifest/pipeline.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ type GitHubProperties struct {
 	// An example for OwnerAndRepository would be: "aws/copilot"
 	OwnerAndRepository    string `structs:"repository" yaml:"repository"`
 	Branch                string `structs:"branch" yaml:"branch"`
-	GithubSecretIdKeyName string `structs:"access_token_secret" yaml:"access_token_secret` // TODO fix naming
```
Comment (iamhopaul123) [inline internal/pkg/manifest/pipeline.go:55]: 🤦‍♂️

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    branches:
+      - '*'
+
+jobs:
+  staticcheck:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v1
```
Comment (iamhopaul123) [inline .github/workflows/ci.yml:14]: why do we want to use v1 instead of v2?

Comment (efekarakus) [reply .github/workflows/ci.yml:14]: I didn't know it existed haha, i'll update :)

File: internal/pkg/cli/task_run.go (hunk 1 above)
Comment (kohidave) [reply internal/pkg/cli/task_run.go:314]: We should add a test for this, then!

Comment (efekarakus) [reply internal/pkg/cli/task_run.go:314]: I looked into it and there is already a test for it, but the error is caught in a separate codepath above while validating the `--default` flag, i.e. this statement cannot be reached :/

I wanted to remove it but it required a refactor on how we validate flags together and wasn't related to the PR so I stopped here.

I'll address the issue when adding the golang-ci-github-action.

**Pull request merged**